pkg/process/process.go

```diff
@@ -0,0 +1,86 @@
+package process
+
+import (
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// EnumProcesses returns a slice containing the process IDs of all processes
+// currently running on the system.
+func EnumProcesses() ([]uint32, error) {
+	count := 256
+	uint32Size := unsafe.Sizeof(uint32(0))
+	for {
+		buf := make([]uint32, count)
+		bufferSize := uint32(len(buf) * int(uint32Size))
+		retBufferSize := uint32(0)
+		if err := enumProcesses(&buf[0], bufferSize, &retBufferSize); err != nil {
+			return nil, err
+		}
+		if retBufferSize == bufferSize {
+			count = count * 2
+			continue
+		}
+		actualCount := retBufferSize / uint32(uint32Size)
+		return buf[:actualCount], nil
+	}
+}
+
+// ProcessMemoryCountersEx is the PROCESS_MEMORY_COUNTERS_EX struct from
+// Windows:
+// https://docs.microsoft.com/en-us/windows/win32/api/psapi/ns-psapi-process_memory_counters_ex
+type ProcessMemoryCountersEx struct {
+	Cb                         uint32
+	PageFaultCount             uint32
+	PeakWorkingSetSize         uint
+	WorkingSetSize             uint
+	QuotaPeakPagedPoolUsage    uint
+	QuotaPagedPoolUsage        uint
+	QuotaPeakNonPagedPoolUsage uint
+	QuotaNonPagedPoolUsage     uint
+	PagefileUsage              uint
+	PeakPagefileUsage          uint
+	PrivateUsage               uint
+}
+
+// GetProcessMemoryInfo returns the memory usage information for the given
+// process. The process handle must have the PROCESS_QUERY_INFORMATION or
+// PROCESS_QUERY_LIMITED_INFORMATION, and the PROCESS_VM_READ access rights.
+func GetProcessMemoryInfo(process windows.Handle) (*ProcessMemoryCountersEx, error) {
+	memCounters := &ProcessMemoryCountersEx{}
+	size := unsafe.Sizeof(*memCounters)
+	if err := getProcessMemoryInfo(process, memCounters, uint32(size)); err != nil {
+		return nil, err
+	}
+	return memCounters, nil
+}
+
+// These constants are used with QueryFullProcessImageName's flags.
+const (
+	// ImageNameFormatWin32Path indicates to format the name as a Win32 path.
+	ImageNameFormatWin32Path = iota
```

Comment:
Dont you need the flag name?

Reply:
What do you mean by flag name?

Reply:
Typically if you wanted to do this you would:
```
type ImageNameFormat int32

const (
    ImageNameFormatWin32Path ImageNameFormat = iota
    ImageNameFormatNTPath
)
```

Without the `typename` I didnt even know this would work. What is the value of `ImageNameFormatNTPath` ? Zero? Shouldnt it be 1?

Reply:
I've seen this kind of format fairly frequently when you don't want to define a full new type (e.g. you want this to be a fairly thin abstraction on the underlying syscall), but do want to define some constants. e.g. golang.org/x/sys/windows has this:

```
const (
    // do not reorder
    NetSetupUnknownStatus = iota
    NetSetupUnjoined
    NetSetupWorkgroupName
    NetSetupDomainName
)
```

The `= iota` should still be inherited by subsequent lines, so `ImageNameFormatNTPath` should be 1.

Reply:
Nvm this is fine. I did not know it worked like this

Reply:
The `iota` expression continues until a new type is encountered so it works with the `default` undefined type as well (int)

Reply:
One note, my understanding is an untyped constant defined as a numerical value will actually not take on any type, rather than defaulting to int. So I think we could actually assign this value to a variable of type `int` or `int8` without a cast.